### PR TITLE
Video-Component

### PR DESCRIPTION
**Pull request by GeronimoJohn**
- Create Video Component
- Migrated VideoJS to Typescript
- Fix Firefox autoplay
- Add stories for Video Component
- Merge main branch to video-component branch

**Video Component**
Remaining TODO: 
Update VideoBlock src/url to be an m3u8/hls file instead of a youtube link (as youtube was out of our scope before). Currently no video will be played if you run journeys until src is changed to an m3u8/hls file. Though a working example can be seen in the storybook.

### CODE REVIEW

File: apps/journeys/src/components/blocks/RadioQuestion/RadioQuestion.tsx [hunk 1]
```diff
@@ -51,7 +51,7 @@ export function RadioQuestion({
           <Typography variant="h1" gutterBottom>
             {label}
           </Typography>
-          <Typography variant="h6">{description}</Typography>
+          <Typography variant="h2">{description}</Typography>
```
Comment (tataihono) [inline apps/journeys/src/components/blocks/RadioQuestion/RadioQuestion.tsx:54]: this change is unrelated. can you revert this and address in a separate PR @GeronimoJohn

File: apps/journeys/src/components/blocks/Video/Video.spec.tsx [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
-    id: 'main',
-    src: 'https://www.youtube.com',
-    title: 'title',
-    parentBlockId: null,
-    provider: VideoProviderEnum.YOUTUBE,
-    children: []
-  }
-  it('should render successfully', () => {
-    const { getByText } = render(<Video {...block} />)
+const block: TreeBlock<VideoBlock> = {
```
Comment (tataihono) [inline apps/journeys/src/components/blocks/Video/Video.spec.tsx:6]: You don't need to specify VideoBlock in here if you don't want to. Merely specifying the `__typename` field does this automatically.

File: apps/journeys/src/components/blocks/Video/Video.tsx [hunk 3]
```diff
@@ -1,8 +1,31 @@
...
-import { TreeBlock } from '../../../libs/transformer/transformer'
 import React, { ReactElement } from 'react'
+import { Container } from '@mui/material'
+import { VideoPlayer } from './VideoPlayer'
+import { TreeBlock } from '../../../libs/transformer/transformer'
 import { GetJourney_journey_blocks_VideoBlock as VideoBlock } from '../../../../__generated__/GetJourney'
 
-export const Video = ({ title }: TreeBlock<VideoBlock>): ReactElement => {
-  return <Box data-testid="Video">Render {title} Here</Box>
+export const Video = ({
```
Comment (tataihono) [inline apps/journeys/src/components/blocks/Video/Video.tsx:7]: This component seems like a pointless wrapper component. VideoPlayer should just replace this video component.

File: apps/journeys/src/components/blocks/Video/Video.tsx [hunk 4]
```diff
@@ -1,8 +1,31 @@
...
+  src,
+  volume = 0,
+  autoplay = false
+}: TreeBlock<VideoBlock>): ReactElement => {
+  const videoJsOptions = {
+    // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
```
Comment (tataihono) [inline apps/journeys/src/components/blocks/Video/Video.tsx:13]: You should use the videojs.PlayerOptions type here. I think that should remove this eslint failure.

File: apps/journeys/src/components/blocks/Video/VideoPlayer.tsx [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+  }, [player, isReady, autoPlaySuccess])
+
+  return (
+    <Container className={classes.container}>
+      <video ref={videoNode} className="video-js" />
+      {children != null ? (
+        <Container className={classes.overlayHolder}>
+          {children as unknown as JSX.Element}
+        </Container>
+      ) : null}
```
Comment (tataihono) [inline apps/journeys/src/components/blocks/Video/VideoPlayer.tsx:109]: can you drop this for now. Deal with overlays in the other branch.

File: apps/journeys/src/components/blocks/Video/VideoPlayer.tsx [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    overlayHolder: {
+      position: 'absolute',
+      top: 0,
+      left: 0,
+      width: '100%',
+      height: 'calc(100% - 30px)',
+      justifyContent: 'center',
+      alignItems: 'center',
+      overflow: 'hidden'
+    }
```
Comment (tataihono) [inline apps/journeys/src/components/blocks/Video/VideoPlayer.tsx:23]: can you drop this for now. Deal with overlays in the other branch.

File: apps/journeys/src/components/blocks/Video/VideoPlayer.tsx [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+  })
+)
+
+interface IVideoPlayerProps {
```
Comment (tataihono) [inline apps/journeys/src/components/blocks/Video/VideoPlayer.tsx:27]:
```suggestion
interface VideoPlayerProps {
```

File: apps/journeys/src/components/blocks/Video/VideoPlayer.tsx [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+  useEffect(() => {
+    if (videoNode.current != null) {
+      player.current = videojs(videoNode.current, {
+        ...initialOptions,
+        ...options
+      })
+      player.current.on('ready', () => {
+        onReady?.(player.current)
+        setIsReady(true)
+      })
```
Comment (tataihono) [inline apps/journeys/src/components/blocks/Video/VideoPlayer.tsx:84]: do you have to remove these listeners as a function of useEffects return value?

File: apps/journeys/src/types.ts [hunk 9]
```diff
@@ -0,0 +1,93 @@
+export type TypeName =
```
Comment (tataihono) [inline apps/journeys/src/types.ts:1]: I don't think this file is needed.

**Review by tataihono — DISMISSED** [review]

File: apps/journeys/src/components/blocks/Video/Video.spec.tsx (hunk 2 above)
Comment (GeronimoJohn) [reply apps/journeys/src/components/blocks/Video/Video.spec.tsx:6]: @tataihono Our other tests have the same pattern, I don't mind changing it, but I think for consistency it might be a good idea to stick with it and follow the pattern we currently have. Unless we're thinking of changing those.

File: apps/journeys/src/components/blocks/Video/Video.tsx (hunk 4 above)
Comment (GeronimoJohn) [reply apps/journeys/src/components/blocks/Video/Video.tsx]: Made some changes - applied your suggestion, sadly error is still there

**Review by tataihono — APPROVED** [review]
